src/python/pants/backend/java/compile/javac.py

```diff
@@ -42,16 +44,25 @@ class CompileJavaSourceRequest(ClasspathEntryRequest):
     field_sets = (JavaFieldSet, JavaGeneratorFieldSet)
 
 
+@dataclass(frozen=True)
+class JavacRequest:
```

Comment:
Should this include command-line flags as well? Whatever produces would fill that in from the applicable subsystem.

Reply:
Also maybe implement `EngineAwareParameter`?

Reply:
Oh `EngineAwareParameter` is super super helpful for debugging because it shows up in stacktraces. You subclass it, then implement `def debug_hint(self) -> str:`. Maybe have it return `self.output_filename`? Or `f"{self.input_filenames} -> {self.output_filename}"`?